pkg/services/pluginsintegration/pluginstore/store_test.go

```diff
@@ -11,45 +11,75 @@ import (
 	"github.com/grafana/grafana/pkg/plugins/backendplugin"
 	"github.com/grafana/grafana/pkg/plugins/log"
 	"github.com/grafana/grafana/pkg/plugins/manager/pluginfakes"
+	"github.com/grafana/grafana/pkg/services/featuremgmt"
 )
 
 func TestStore_ProvideService(t *testing.T) {
 	t.Run("Plugin sources are added in order", func(t *testing.T) {
-		var loadedSrcs []plugins.Class
-		l := &pluginfakes.FakeLoader{
-			LoadFunc: func(ctx context.Context, src plugins.PluginSource) ([]*plugins.Plugin, error) {
-				loadedSrcs = append(loadedSrcs, src.PluginClass(ctx))
-				return nil, nil
+		tests := []struct {
+			name           string
+			featureEnabled bool
+			expectedOrder  []plugins.Class
+		}{
+			{
+				name:           "with FlagPluginStoreServiceLoading disabled",
+				featureEnabled: false,
+				expectedOrder:  []plugins.Class{"1", "2", "3"},
+			},
+			{
+				name:           "with FlagPluginStoreServiceLoading enabled",
+				featureEnabled: true,
+				expectedOrder:  []plugins.Class{"1", "2", "3"},
```

Comment:
Should we not be testing that `ProvideService` with feature enabled won't call sources/the loader?

Reply:
@wbrowne i updated the test. let me know if it covers what you were thinking here